barretenberg/cpp/src/barretenberg/dsl/acir_format/ivc_recursion_constraint.cpp

```diff
@@ -106,26 +106,35 @@ ClientIVC::VerifierInputs create_mock_verification_queue_entry(const ClientIVC::
     blocks.compute_offsets(/*is_structured=*/true);
     size_t dyadic_size = blocks.get_structured_dyadic_size();
     size_t pub_inputs_offset = blocks.pub_inputs.trace_offset();
-    // All circuits have pairing point public inputs; kernels have additional public inputs for two databus commitments
-    size_t num_public_inputs = bb::PAIRING_POINTS_SIZE;
-    if (is_kernel) {
-        num_public_inputs += bb::PROPAGATED_DATABUS_COMMITMENTS_SIZE;
-    }
 
-    // Construct a mock Oink or PG proof
+    // Construct a mock Oink or PG proof and a mock MegaHonk verification key
     std::vector<FF> proof;
-    if (verification_type == ClientIVC::QUEUE_TYPE::OINK) {
-        proof = create_mock_oink_proof<ClientIVC::Flavor>(num_public_inputs);
-    } else if (verification_type == ClientIVC::QUEUE_TYPE::PG || verification_type == ClientIVC::QUEUE_TYPE::PG_FINAL) {
-        proof = create_mock_pg_proof<ClientIVC::Flavor>(num_public_inputs);
+    std::shared_ptr<MegaVerificationKey> verification_key;
+
+    if (is_kernel) {
```

Comment:
please use a switch statement on queue type in the two branches since it's an enum

Reply:
using Flavor = ClientIVC::Flavor;
using KernelIO = ... in one branch
using AppIO = ... in the other branche